src/SprykerSdk/Zed/ComposerConstrainer/Business/Composer/ComposerJsonWriter.php

```diff
@@ -31,6 +31,9 @@ public function __construct(ComposerConstrainerConfig $config)
      */
     public function write(array $composerJsonArray): bool
     {
-        return (bool)file_put_contents($this->config->getProjectRootPath() . 'composer.json', json_encode($composerJsonArray, JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES));
+        $encodedJson4Spaces = json_encode($composerJsonArray, JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES);
+        $encodedJson2Spaces = preg_replace('/^(  +?)\\1(?=[^ ])/m', '$1', $encodedJson4Spaces) . "\n";
```

Comment:
IMO it would be great if we can move regex to constant. But this is not a blocker

Reply:
There is no value in that IMO, and decreases readability in favor of reusability/customization that we dont need in this case.